### ISSUE TITLE
dotnet test cache/result output cache

### ISSUE THREAD

**Issue by andresrsanchez** [dotnet/sdk#41589]
### Is your feature request related to a problem? Please describe.
Hi, i would like to know the options that we currently have to cache test results.
**Scenario**: Github ARC (EphemeralRunners) and dotnet test, if your host disappear (spot instances) how can you retry only de failed/not executed tests?

- In PHPUnit they have `cacheResults`.
- In golang since 1.10 the default behaviour is to cache tests.
- In python they have `pytest-cache`

What are our options in dotnet? Is the `filter` our only chance to try to hack it and get the same behaviour? Because i don't know the performance of that when you need to skip lets say 5k tests. 

### Describe the solution you'd like
A clear way to get the previous executed/failed tests and skip or not them, basically get the previous test output state.

### Additional context
Somehow related https://github.com/dotnet/sdk/issues/8643

Thanks!

**Comment by nohwnd** [comment]
Yes filter can be used for this. There is also translation layer that gives you programatic api to communicate with the test runner. There you can collect all the failed tests and send them for execution. E.g. how VS would do it. 

https://github.com/microsoft/vstest/blob/main/playground/TestPlatform.Playground/Program.cs this project is a good place to start experimenting with the translation layer of vstest.
